sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Metrics/ExecutableLinesWalker.cs

```diff
@@ -29,15 +32,23 @@ public class ExecutableLinesWalker : CSharpSyntaxWalker
     {
         private readonly HashSet<int> executableLineNumbers = new HashSet<int>();
 
+        private static readonly string[] ExcludeFromCoverageAttributes =
+            {
+                "ExcludeFromCodeCoverage",
+                "ExcludeFromCodeCoverageAttribute"
+            };
+
         public ICollection<int> ExecutableLines => executableLineNumbers;
 
         public override void DefaultVisit(SyntaxNode node)
         {
-            FindExecutableLines(node);
-            base.DefaultVisit(node);
+            if (FindExecutableLines(node))
+            {
+                base.DefaultVisit(node);
+            }
         }
 
-        private void FindExecutableLines(SyntaxNode node)
+        private bool FindExecutableLines(SyntaxNode node)

```

Comment:
The name of the method compared to its behavior is not so nice but I cannot find any better.